### PR TITLE
Bearer tokens

### PR DESCRIPTION
**Pull request by betaredex**
Contains everything from https://github.com/FNNDSC/pfcon/pull/26 with some modifications:
- Set up the openshift secret correctly
- Moved Auth.py to pfmisc
- Disable token authentication by default
- Update the swift openshift template

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -15,11 +15,11 @@
 #
 # To run an interactive shell inside this container, do:
 #
-#   docker run -ti --rm --entrypoint /bin/bash local/pfioh
+#   docker run -ti --entrypoint /bin/bash local/pfioh
```
Comment (danmcp) [inline Dockerfile]: I don't think you meant to do this or the one below.

File: README.rst [hunk 2]
```diff
@@ -87,7 +87,7 @@ This docker has an entry point ``python3``. To enter the dock at a different ent
 
 .. code-block:: bash
 
-   docker run --rm -ti --entrypoint /bin/bash fnndsc/ubuntu-python3
```
Comment (danmcp) [inline README.rst]: or this one.

File: pfioh/_colors.py [hunk 3]
```diff
@@ -58,4 +58,4 @@ def strip( text ):
 
         for c in members:
             text = text.replace( vars( Colors )[c], '' )
-        return text
+        return text
```
Comment (danmcp) [inline pfioh/_colors.py]: why?

File: pfioh/mount_dir.py [hunk 4]
```diff
@@ -91,7 +91,8 @@ def getData(self, **kwargs):
             self.dp.qprint("Zipping target '%s'..." % str_fileToProcess, comms = 'status')
  
             str_dirSuffix   = ""
-            if os.path.isdir(str_fileToProcess):
+            # Ensure that directory paths end in '/'
+            if os.path.isdir(str_fileToProcess) and str_fileToProcess[-1] != '/':
```
Comment (danmcp) [inline pfioh/mount_dir.py]: @jdtzmn Just fixed this another way.

File: pfioh/pfioh.py [hunk 5]
```diff
@@ -818,7 +871,7 @@ def do_GET_preop(self, **kwargs):
                 b_status    = True
                 d_ret['cmd']    = str_cmd
             if 'op' in d_preop.keys():
-                # pudb.set_trace()
+                #
```
Comment (danmcp) [inline pfioh/pfioh.py]: Why leave the #

File: pfioh/pfioh.py [hunk 6]
```diff
@@ -1089,6 +1142,7 @@ def __init__(self, *args, **kwargs):
 
         :return:
         """
+
```
Comment (danmcp) [inline pfioh/pfioh.py]: Why?

File: pfioh/pfioh.py [hunk 7]
```diff
@@ -1123,24 +1176,31 @@ def setup(self, **kwargs):
...
         self.col2_print("Listening on address:",    self.args['ip'])
         self.col2_print("Listening on port:",       self.args['port'])
         self.col2_print("Server listen forever:",   self.args['b_forever'])
         self.col2_print("Return HTTP responses:",   self.args['b_httpResponse'])
+        self.col2_print("Authorization enabled:",   self.args['b_tokenAuth'])
 
         print(Colors.LIGHT_GREEN + "\n\n\tWaiting for incoming data..." + Colors.NO_COLOUR, flush=True)
 
 
+
```
Comment (danmcp) [inline pfioh/pfioh.py]: Why?

File: example-secret.yml [hunk 8]
```diff
@@ -0,0 +1,7 @@
+apiVersion: v1
```
Comment (danmcp) [inline example-secret.yml]: @betaredex Same comment here as in pfioh.  I think the readme needs updates based on these changes.

**Comment by danmcp** [comment]
@betaredex This PR has conflicts in it that needs to be resolved with upstream.

File: example-secret.yml (hunk 8 above)
Comment (betaredex) [reply example-secret.yml]: Will do.

File: pfioh/pfioh.py [hunk 9]
```diff
@@ -994,7 +1070,7 @@ def do_POST_withCompression(self, **kwargs):
         d_ret= self.storeData(file_name= str_localFile, client_path = str_fileName,
                               file_content= data, Path= str_unpackPath, is_zip=b_zip,d_ret=d_ret)
         
-        # pudb.set_trace()
+        #
```
Comment (danmcp) [inline pfioh/pfioh.py]: Why leave the # ?

File: pfioh/pfioh.py [hunk 10]
```diff
@@ -263,7 +287,41 @@ def log_message(self, format, *args):
         return
 
     def do_GET(self):
+        isAuthorized, error = self.authorizeRequest()
+        if isAuthorized:
+            return self.execute_GET()
+        else:
+            d_ret = {
```
Comment (danmcp) [inline pfioh/pfioh.py]: This looks like the same code from do_POST

File: bin/pfioh [hunk 11]
```diff
@@ -298,6 +311,7 @@ parser.add_argument(
     default = False
 ) 
 
+
```
Comment (danmcp) [inline bin/pfioh]: why?

File: pfioh/pfioh.py (hunk 10 above)
Comment (betaredex) [reply pfioh/pfioh.py]: No, it runs execute_GET instead of execute_POST (Emilio's code)

Comment (danmcp) [reply pfioh/pfioh.py]: I am talking about the else block.  It looks like there is a good use case for code reuse in a shared method.

**Pull request merged**